### ISSUE TITLE
Misc: To release 1.0.26

### ISSUE THREAD

**Issue by tormodvolden** [libusb/libusb#1047]
Things we want to push first, just after the 1.0.25 release (potentially for fix-up release or at least straight-forward cherry-picking for downstreams), in order:

Should also be safe and useful for a fix-up release:

Ready to merge:
 
Must be checked:

_Everything else is postponed to 1.0.27 (see 1094) because 1.0.26 will be for regression, critical or build fixes only._

Merged regression and build fixes:
- #1073
- #1058 workaround for outdated pyusb and some application bugs now exposed
- #1089 temporarily unbreak broken applications
- #1082
- #1095 regression in 1.0.24
- #1070 fix build on older macOS
- #1064 fix build on sparc
- #1062 fix warning with -O3 that prevents -Wall builds
- #1068 MinGW binaries without dependencies
- darwin: fix build warning from #1035
- #1097

Merged features and improvements, regular bug fixes:
- #1119
- #1093 avoid crash with broken device
- #1091 fix Product string retrieval on Windows HID
- #1088
- #1078 device emulation tests
- #1060 improve Windows reporting
- #1056 use newer non-deprecated API consistently on darwin
- #1071 comment fixes only
- #1107
- #1110 (documentation update)

**Comment by mcuee** [comment]
hotplug is the main feature missing for Windows. However, if we do not have Chris back in the project or another Windows developer to chime in, I think this will not happen in libusb-1.0.26 either.
* https://github.com/libusb/libusb/issues/86

**Comment by mcuee** [comment]
This is probably good to be in libusb 1.0.26 as well.
* https://github.com/libusb/libusb/issues/1048

**Comment by tormodvolden** [comment]
We should for a while after the 1.0.25 release avoid larger changes and rabbit holes that prevent us from releasing any regression fixes in due time. Or we have to use stable branches, but then we'd have even more to maintain and test.

For something like the Windows hotplug support we must be sure there are developers around that will stay available to timely fix any issues that arises after it lands. Anyway, until there is at least a draft of a pull request for it, it remains just a wishlist item with no relevance for our release milestones.

**Comment by mcuee** [comment]
> For something like the Windows hotplug support we must be sure there are developers around that will stay available to timely fix any issues that arises after it lands. Anyway, until there is at least a draft of a pull request for it, it remains just a wishlist item with no relevance for our release milestones.

Good point.

**Comment by tormodvolden** [comment]
It seems like we should pretty soon release a 1.0.26 with only regression and build fixes, and leave API changes and more complicated changes for 1.0.27. Anyway we can still wait a bit to see if other regressions pop up (like the hotplug issue reported today).

**Comment by mcuee** [comment]
I agree to release a 1.0.26 version with only regression and build fixes.

**Comment by mcuee** [comment]
Thanks for the work done by @tormodvolden, we are closer to 1.0.26rc1 now.

**Comment by tormodvolden** [comment]
More accurately, @benzea did all the work :) I think I will merge the "safe" items on the list as well, even if they are not strictly critical bug fixes. Possibly tag the RC in a few days, and if everything is good, release it around March 31 (two months after 1.0.25 already).

**Comment by mcuee** [comment]
I also agree that #1071 and #1078 are ready to be merged. I like #1078 very much. It can be a highlight of 1.0.26 as a new feature.

**Comment by lunixbochs** [comment]
Can I request my PR https://github.com/libusb/libusb/pull/1093 lands in the RC, as it's a memory safety / crash issue?

**Comment by mcuee** [comment]
* #1093 has been merged.

**Comment by tormodvolden** [comment]
With the RC out today, we can aim for final release around April 10th, if nothing complicated turns up. I hope to test some libkinect isochronous streaming with XBOX before then :)

**Comment by mcuee** [comment]
#1102 should be safe to merge.

**Comment by mcuee** [comment]
* https://github.com/libusb/libusb/pull/1101 seems to be okay as well.

**Comment by mcuee** [comment]
So the main remaining thing is to merge #1119 and deal with #1110 -- both are for macOS.

**Comment by sergeu90** [comment]
Hi all. a lot of thanks for library. When do you plan release 1.0.26 version? I saw in this conversation 10 april?

**Comment by mcuee** [comment]
#1119 has been merged. Only pending #1110.

**Comment by tormodvolden** [comment]
Any dates given are approximate and just estimation. But we are still on track :)

**Comment by tormodvolden** [comment]
Like last time we will have a short post-release freeze before starting merging new things, just in case something critical shows up after release.

**Comment by mcuee** [comment]
Two weeks freeze?
